packages/core/FuzzedDataProvider.ts

```diff
@@ -372,33 +372,89 @@ export class FuzzedDataProvider {
 	 * is not sufficiently long.
 	 * @param maxLength the maximum length of the string
 	 * @param encoding the encoding of the string
+	 * @param printable - a boolean, which defaults to false that indicates whether consumed strings
+	 * should be forced to contain only valid printable characters
 	 * @returns a `string` of length between 0 and `maxLength` (inclusive)
 	 */
 	consumeString(
 		maxLength: number,
-		encoding: BufferEncoding | undefined = "ascii"
+		encoding: BufferEncoding | undefined = "ascii",
+		printable: boolean | undefined = false
 	): string {
 		if (maxLength < 0) throw new Error("maxLength must be non-negative");
+		let result;
 		const arrayLength = Math.min(maxLength, this._remainingBytes);
-		const result = this.data.toString(
-			encoding,
-			this.dataPtr,
-			this.dataPtr + arrayLength
-		);
+
+		if (printable) {
+			result = this.bufToPrintableString(
+				this.data,
+				this.dataPtr,
+				this.dataPtr + arrayLength,
+				encoding
+			);
+		} else {
+			result = this.data.toString(
+				encoding,
+				this.dataPtr,
+				this.dataPtr + arrayLength
+			);
+		}
 		this.dataPtr += arrayLength;
 		this._remainingBytes -= arrayLength;
 		return result;
 	}
 
+	/**
+	 * A simple conversion helper that yields the ASCII representaion of a character.
+	 * @param char is a single character from a string represented as a number
+	 * @returns a 'string' from a single character that was represented by a number
+	 */
+	toAscii(char: number): number {
+		if (char < 32 || char > 126) {
+			const moduloInRange = (char - 32 + 256) % 256;
+			return (moduloInRange % 95) + 32;
+		}
+		return char;
+	}
+
+	/**
+	 * Helper function that converts the given string type into one that only
+	 * contains printable characters. This conversion produces seemingly arbitrary
```

Comment:
> This conversion produces seemingly arbitrary (non-smart) but printable strings, IFF the entropy of the input buffer is high enough.

I don't understand what that means. Can you explain?

Reply:
The generated Strings are "dumb" as in the conversion only attempts to convert a byte buffer to a printable string. If the buffer e.g. looks like this `[0, 0, 0, 0, 0, ....]` the entropy is low and therefore the applied conversion formula will produce something like `BBBBB...`. So in essence, the comment just states that the quality of the output is highly dependent on the given buffer.

Reply:
I think, it's a bit vague. After double-checking, given a uniform distribution of the input, the output distribution will be as follows:
1) numbers [32; 97] will have the probability of about 0.01172 of occuring,
2) numbers [98; 126] will have probability of 0.00781 of occurring.
Feel free to add it (and double check!)

Reply:
The new approach has the exact same mapping. But I like the lookup table solution---it should be more performant than computing all those operations on the fly, which Javascript does using 64-bit floats :wink: 

Can you make the lookup functional so that it does not modify the original buffer?

> The quality of the output heaviliy depends on the input buffer entropy.

What does "quality" mean here? The reason I don't like this comment is because it's probably true for most functions that process input. Also, here the user does not have any influence on the input because it comes from the fuzzer. But tell me that numbers 32 to 97 are slightly more likely to occur than other printable numbers, then I know what to expect!